### ISSUE TITLE
Failing install

### ISSUE THREAD

**Issue by muschellij2** [jonclayden/RNifti#16]
On mac OSX:
```r
Loading required package: colorout
* installing *source* package ‘RNifti’ ...
** package ‘RNifti’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
ccache clang++ -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -DNDEBUG -DHAVE_ZLIB -DUSING_R -I../inst/include -Izlib -I"/Library/Frameworks/R.framework/Versions/3.6/Resources/library/Rcpp/include" -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include  -fPIC  -Wall -g -O2  -c main.cpp -o main.o
main.cpp:511:28: error: default initialization of an object of const type 'const Rcomplex' without a user-provided default constructor
            const Rcomplex naValue { NA_REAL, NA_REAL };
                           ^
                                   = {}
main.cpp:511:35: error: expected ';' at end of declaration
            const Rcomplex naValue { NA_REAL, NA_REAL };
                                  ^
                                  ;
main.cpp:565:28: error: default initialization of an object of const type 'const Rcomplex' without a user-provided default constructor
            const Rcomplex naValue { NA_REAL, NA_REAL };
                           ^
                                   = {}
main.cpp:565:35: error: expected ';' at end of declaration
            const Rcomplex naValue { NA_REAL, NA_REAL };
                                  ^
                                  ;
4 errors generated.
make: *** [main.o] Error 1
ERROR: compilation failed for package ‘RNifti’
* removing ‘/Library/Frameworks/R.framework/Versions/3.6/Resources/library/RNifti’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/3.6/Resources/library/RNifti’
Warning in install.packages :
  installation of package ‘RNifti’ had non-zero exit status
```

<summary>Session info</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ──────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.6.0 (2019-04-26)
#>  os       macOS Mojave 10.14.6        
#>  system   x86_64, darwin15.6.0        
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/New_York            
#>  date     2019-11-21                  
#> 
#> ─ Packages ──────────────────────────────────────────────────────────────
#>  package     * version date       lib source                            
#>  assertthat    0.2.1   2019-03-21 [1] CRAN (R 3.6.0)                    
#>  cli           1.1.0   2019-03-19 [1] CRAN (R 3.6.0)                    
#>  crayon        1.3.4   2017-09-16 [1] CRAN (R 3.6.0)                    
#>  digest        0.6.22  2019-10-21 [1] CRAN (R 3.6.0)                    
#>  evaluate      0.14    2019-05-28 [1] CRAN (R 3.6.0)                    
#>  highr         0.8     2019-03-20 [1] CRAN (R 3.6.0)                    
#>  htmltools     0.4.0   2019-10-04 [1] CRAN (R 3.6.0)                    
#>  knitr         1.24.3  2019-08-28 [1] Github (muschellij2/knitr@abcea3d)
#>  magrittr      1.5     2014-11-22 [1] CRAN (R 3.6.0)                    
#>  Rcpp          1.0.3   2019-11-08 [1] CRAN (R 3.6.0)                    
#>  rlang         0.4.1   2019-10-24 [1] CRAN (R 3.6.0)                    
#>  rmarkdown     1.16    2019-10-01 [1] CRAN (R 3.6.0)                    
#>  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.6.0)                    
#>  stringi       1.4.3   2019-03-12 [1] CRAN (R 3.6.0)                    
#>  stringr       1.4.0   2019-02-10 [1] CRAN (R 3.6.0)                    
#>  withr         2.1.2   2018-03-15 [1] CRAN (R 3.6.0)                    
#>  xfun          0.11    2019-11-12 [1] CRAN (R 3.6.0)                    
#>  yaml          2.2.0   2018-07-25 [1] CRAN (R 3.6.0)                    
#> 
#> [1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library
```

</details>

**Comment by muschellij2** [comment]
Adding `-std=c++11` to my CXX worked fine:

My `~/.R/Makevars` was:
```
CCACHE=ccache
CC=$(CCACHE) clang
CXX=$(CCACHE) clang++
CXX11=$(CCACHE) clang++

SHLIB_OPENMP_CFLAGS=
SHLIB_OPENMP_CXXFLAGS=
SHLIB_OPENMP_CPPFLAGS=
SHLIB_OPENMP_FCFLAGS= 
SHLIB_OPENMP_FFLAGS=
```

but Adding 
```
CXX=$(CCACHE) clang++ -std=c++11
```
works fine.

**Comment by jonclayden** [comment]
Good catch John – thanks. My `clang++` seems to use C++11 by default if no standard is specified, which I hadn’t realised. Anyway, this use of C++11 syntax is inadvertent and easily avoided, so I will modify these lines rather than changing the whole package’s system requirements.

**Comment by jonclayden** [comment]
Fixed minimally in commit bb77991.
